cpp/src/strings/convert/convert_durations.cu

```diff
@@ -423,30 +414,16 @@ struct dispatch_from_durations_fn {
 
     // copy null mask
     rmm::device_buffer null_mask = cudf::detail::copy_bitmask(durations, stream, mr);
-    // build offsets column
-    auto offsets_transformer_itr = thrust::make_transform_iterator(
-      thrust::make_counting_iterator<int32_t>(0),
-      duration_to_string_size_fn<T>{d_column, d_format_items, compiler.items_count()});
-    auto offsets_column = detail::make_offsets_child_column(
-      offsets_transformer_itr, offsets_transformer_itr + strings_count, stream, mr);
-    auto offsets_view  = offsets_column->view();
-    auto d_new_offsets = offsets_view.template data<int32_t>();
-
-    // build chars column
-    auto const chars_bytes =
-      cudf::detail::get_value<int32_t>(offsets_column->view(), strings_count, stream);
-    auto chars_column = detail::create_chars_child_column(chars_bytes, stream, mr);
-    auto d_chars      = chars_column->mutable_view().template data<char>();
-
-    thrust::for_each_n(rmm::exec_policy(stream),
-                       thrust::make_counting_iterator<size_type>(0),
-                       strings_count,
-                       duration_to_string_fn<T>{
-                         d_column, d_format_items, compiler.items_count(), d_new_offsets, d_chars});
+
+    auto children =
+      make_strings_children(from_durations_fn<T>{d_column, d_format_items, compiler.items_count()},
+                            strings_count,
+                            stream,
+                            mr);
 
     return make_strings_column(strings_count,
-                               std::move(offsets_column),
-                               std::move(chars_column),
+                               std::move(children.first),
+                               std::move(children.second),
```

Comment:
Question: Would this benefit from using structured bindings?

Reply:
yup! updated